apm-agent-core/src/main/java/co/elastic/apm/agent/impl/ElasticApmTracer.java

```diff
@@ -383,20 +384,27 @@ public void endSpan(Span span) {
             span.decrementReferences();
             return;
         }
-        if (!span.isComposite()) {
+        if (span.isExit()) {
+            SpanConfiguration spanConfiguration = getConfig(SpanConfiguration.class);
```

Comment:
Store spanConfiguration in a field so that we don't have to look up the config on each span end.

Reply:
Should I do the same for `coreConfiguration.getSpanMinDuration()`?

Reply:
It's not strictly necessary but a field lookup is cheaper than a map lookup and it does not increase the complexity in any way. This was not meant as a blocker to this PR but a suggestion for a minor improvement 🙂 

The `coreConfiguration` is already a field, so no changes needed there, I think.

Reply:
what looks a little more worrying is that the map is a hashmap rather than a concurrent map. It looks like the registration is all done first before accessing, so this should be fine but I'm not sure it's guaranteed

Reply:
@tobiasstadler may as well make the 2 configs consistent, ie both fields, before we merge this

Reply:
Done